frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/toolbar/RebaseCurrentBranchOntoParentAction.java

```diff
@@ -33,25 +33,28 @@ public void update(AnActionEvent anActionEvent) {
     super.update(anActionEvent);
 
     Presentation presentation = anActionEvent.getPresentation();
-    if (presentation.isEnabledAndVisible()) {
-      var currentBranch = getGitMacheteRepository(anActionEvent).flatMap(repository -> repository.getCurrentBranchIfManaged());
+    if (!presentation.isEnabledAndVisible()) {
+      return;
+    }
+
+    var currentBranch = getGitMacheteRepository(anActionEvent)
+        .flatMap(repository -> repository.getCurrentBranchIfManaged());
 
-      if (currentBranch.isEmpty()) {
-        presentation.setDescription("Current revision is not a branch managed by Git Machete");
-        presentation.setEnabled(false);
+    if (currentBranch.isEmpty()) {
+      presentation.setDescription("Current revision is not a branch managed by Git Machete");
+      presentation.setEnabled(false);
 
-      } else if (currentBranch.get().isRootBranch()) {
-        presentation.setDescription("Can't rebase root branch '${currentBranch.get().getName()}'");
-        presentation.setEnabled(false);
+    } else if (currentBranch.get().isRootBranch()) {
+      presentation.setDescription("Can't rebase root branch '${currentBranch.get().getName()}'");
+      presentation.setEnabled(false);
 
-      } else if (currentBranch.get().asNonRootBranch().getSyncToParentStatus().equals(SyncToParentStatus.Merged)) {
-        presentation.setEnabled(false);
-        presentation.setDescription("Can't rebase merged branch '${currentBranch.get().getName()}'");
+    } else if (currentBranch.get().asNonRootBranch().getSyncToParentStatus().equals(SyncToParentStatus.Merged)) {
+      presentation.setEnabled(false);
+      presentation.setDescription("Can't rebase merged branch '${currentBranch.get().getName()}'");
 
-      } else if (currentBranch.get().isNonRootBranch()) {
-        var upstreamBranch = currentBranch.get().asNonRootBranch().getUpstreamBranch();
-        presentation.setDescription("Rebase '${currentBranch.get().getName()}' onto '${upstreamBranch.getName()}'");
-      }
+    } else if (currentBranch.get().isNonRootBranch()) {
```

Comment:
Fun fact: if you mark both `BaseGitMacheteNonRootBranch#getSyncToParentStatus` and `BaseGitMacheteBranch#asNonRootBranch` as `@SideEffectFree`, this is no longer necessary (since you declare both methods as non-side-effecting, Checker can assume that "non-rootness" of `currentBranch.get()` still holds at this point. Up to you, same in `RebaseSelectedBranchOntoParentAction`

Reply:
I leave it like it is as an additional check anyway